### PR TITLE
Bug 1141887 - [Stingray] apply focus manager to app window related UIs

### PR DESCRIPTION
**Pull request by huchengtw-moz**

### CODE REVIEW

**Comment by try-server-hook** [comment]
[huchengtw-moz John Hu (huchengtw-moz)](https://github.com/huchengtw-moz) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=c2f4f06d4a5b)

**Comment by try-server-hook** [comment]
[huchengtw-moz John Hu (huchengtw-moz)](https://github.com/huchengtw-moz) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=1e1e01ded3df)

**Comment by try-server-hook** [comment]
[huchengtw-moz John Hu (huchengtw-moz)](https://github.com/huchengtw-moz) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=b66a34a5dfb0)

**Comment by try-server-hook** [comment]
[huchengtw-moz John Hu (huchengtw-moz)](https://github.com/huchengtw-moz) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=ada76859cfc0)

**Comment by try-server-hook** [comment]
[huchengtw-moz John Hu (huchengtw-moz)](https://github.com/huchengtw-moz) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=bb7765e4b437)

**Comment by try-server-hook** [comment]
[huchengtw-moz John Hu (huchengtw-moz)](https://github.com/huchengtw-moz) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=09fa3c6bbd95)

**Comment by try-server-hook** [comment]
[huchengtw-moz John Hu (huchengtw-moz)](https://github.com/huchengtw-moz) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=7218df8f03bb)

File: tv_apps/smart-system/js/action_menu.js [hunk 1]
```diff
@@ -209,7 +218,7 @@
      * @return {Boolean} if the container is invisible, return false
      * @return {Number} if the container is visible, return z-index
```
Comment (luke-chang) [inline tv_apps/smart-system/js/action_menu.js]: Please fix the comment above as well.

File: tv_apps/smart-system/js/action_menu.js [hunk 2]
```diff
@@ -209,7 +218,7 @@
...
      */
-    isVisible: function() {
+    isFocusable: function() {
       return this.container && this.container.classList.contains('visible');
```
Comment (luke-chang) [inline tv_apps/smart-system/js/action_menu.js]: Seems `this.visible` can be reused here: `return this.container && this.visible;`

File: tv_apps/smart-system/js/app_authentication_dialog.js [hunk 3]
```diff
@@ -210,8 +211,8 @@
    * Determine whether the dialog is visible.
```
Comment (luke-chang) [inline tv_apps/smart-system/js/app_authentication_dialog.js]: ... and fix here.

File: tv_apps/smart-system/js/focus_manager.js [hunk 4]
```diff
@@ -34,12 +36,19 @@
...
+        this.startFallbackCatcher();
+      }
+    }.bind(this));
   };
 
   proto.stop = function ff_stop() {
     window.removeEventListener('screenchange', this);
-    this.stopFallbackCatcher();
+    if (this._autoFixingEnabled) {
+      this.stopFallbackCatcher();
```
Comment (luke-chang) [inline tv_apps/smart-system/js/focus_manager.js]: I'd prefer resetting `this._autoFixingEnabled` to `false` to avoid exceptions if any.

**Comment by try-server-hook** [comment]
[huchengtw-moz John Hu (huchengtw-moz)](https://github.com/huchengtw-moz) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=60ff2433642e)

**Pull request merged**